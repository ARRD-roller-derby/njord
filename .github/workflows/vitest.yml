name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30