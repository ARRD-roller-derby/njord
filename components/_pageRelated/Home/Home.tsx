import AuthentificatedLayout from "../../_layouts/Authentificated/Authentificated";
import classes from "./Home.module.css";
import TrainingCard from "../../Events/Training/TrainingCard/TrainingCard";
import dayjs from 'dayjs'

const fakeTraining = [

  {
    id:1,
    type:"training",
    date: dayjs().add(5,'days'),
    startHour:'17:00',
    endHour:'19:30',

  }
]
export default function Home() {

  return (
    <AuthentificatedLayout>
      <div className={classes.container}>
      <h2>chips</h2>
      <div className={classes.events}>
        {fakeTraining.map(event=> <TrainingCard key={event.id} training={event}/>)}
      </div>
      <p>Message si aucun event</p>
      </div>
    </AuthentificatedLayout>
  );
}
