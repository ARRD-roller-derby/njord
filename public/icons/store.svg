<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   viewBox="0 0 576 512"
   version="1.1"
   id="svg10"
   sodipodi:docname="store-duotone.svg"
   inkscape:version="1.0.1 (3bc2e813f5, 2020-09-07)">
  <metadata
     id="metadata14">
    <rdf:RDF>
      <cc:Work
         rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type
           rdf:resource="http://purl.org/dc/dcmitype/StillImage" />
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <sodipodi:namedview
     pagecolor="#ffffff"
     bordercolor="#666666"
     borderopacity="1"
     objecttolerance="10"
     gridtolerance="10"
     guidetolerance="10"
     inkscape:pageopacity="0"
     inkscape:pageshadow="2"
     inkscape:window-width="1920"
     inkscape:window-height="1009"
     id="namedview12"
     showgrid="false"
     inkscape:zoom="0.83984375"
     inkscape:cx="329.78442"
     inkscape:cy="254.52185"
     inkscape:window-x="1272"
     inkscape:window-y="-8"
     inkscape:window-maximized="1"
     inkscape:current-layer="svg10" />
  <!--! Font Awesome Pro 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2022 Fonticons, Inc. -->
  <defs
     id="defs4">
    <style
       id="style2">.fa-secondary{opacity:.4}</style>
  </defs>
  <path
     class="fa-primary"
     d="M490.3 13.13L547.6 103.8C577.3 150.7 551 215.8 495.5 223.2C491.6 223.7 487.6 224 483.4 224C457.4 224 434.2 212.6 418.3 195C402.4 212.6 379.2 224 353.1 224C327 224 303.8 212.6 287.9 195C272 212.6 248.9 224 222.7 224C196.7 224 173.5 212.6 157.6 195C141.7 212.6 118.5 224 92.36 224C88.3 224 84.21 223.7 80.24 223.2C24.92 215.8-1.255 150.6 28.33 103.8L85.66 13.13C90.76 4.979 99.87 0 109.6 0H466.4C476.1 0 485.2 4.978 490.3 13.13H490.3z"
     id="path6"
     style="fill:#999999" />
  <path
     class="fa-secondary"
     d="M80.23 223.2C84.21 223.7 88.3 224 92.36 224C105.1 224 117.2 221.3 128 216.5V384H448V216.5C458.8 221.3 470.8 224 483.4 224C487.6 224 491.6 223.7 495.5 223.2C501.4 222.4 506.9 221 512 219.1V448C512 483.3 483.3 512 448 512H128C92.65 512 64 483.3 64 448V219.1C69.08 221 74.5 222.4 80.24 223.2H80.23z"
     id="path8"
     style="fill:#b3b3b3" />
</svg>
