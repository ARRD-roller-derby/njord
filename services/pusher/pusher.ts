import Pusher from 'pusher'

const pusherConnect = new Pusher({
  appId: process.env.PUSHER_APP_ID,
  key: process.env.NEXT_PUBLIC_PUSHER_APP_KEY,
  secret: process.env.PUSHER_APP_SECRET,
  cluster: process.env.NEXT_PUBLIC_PUSHER_CLUSTER,
})

declare global {
  var pusher: any | undefined
}

export const pusher = global.pusher || pusherConnect
if (process.env.NODE_ENV !== 'production') pusher.pusher = pusherConnect
