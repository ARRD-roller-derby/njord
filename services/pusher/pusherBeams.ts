import PushNotifications from '@pusher/push-notifications-server'

const pusherBeams = new PushNotifications({
  instanceId: process.env.PUSHER_INSTANCE_ID,
  secretKey: process.env.PUSHER_SECRET_KEY,
})

declare global {
  var pushNotifications: any | undefined
}

export const pushNotifications = global.pushNotifications || pusherBeams
if (process.env.NODE_ENV !== 'production') pushNotifications.pushNotifications = pusherBeams
