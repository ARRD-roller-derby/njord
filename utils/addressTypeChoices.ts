import { addressType } from "../types/addressType.enum";

export const addressTypeChoices = [
  {
    label: 'Mon adresse',
    value:addressType.personnal
  },
  {
    label: 'Stade',
    value:addressType.stadium
  },
  {
    label: 'Point de covoiturage',
    value:addressType.carpool
  },
  {
    label: 'Autre',
    value:addressType.other
  }
]