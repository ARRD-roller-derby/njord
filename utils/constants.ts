export const DISCORD_NEWS_HOOK = process.env.DISCORD_NEWS_HOOK || ''