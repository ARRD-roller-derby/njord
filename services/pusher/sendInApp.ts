import { pusher } from './pusher'

// if change service, just change that.
export default function sendInApp(channel: string, payload: Object): void {
  pusher.trigger(channel, 'message', payload)
}
