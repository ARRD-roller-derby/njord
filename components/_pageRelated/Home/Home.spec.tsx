import { afterEach, describe, expect, it, vi } from 'vitest'
import {cleanup } from '@testing-library/react'
import Home from './Home'
import { PusherContext } from '../../../stores/pusher.store'
import { render, screen } from '../../../utils/test-utils'

describe('<Home />', () => {
  afterEach(cleanup)

  it('Check snapshot', async () => {
    const { asFragment } = render(
      <PusherContext.Provider value={[null, vi.fn]}>
        <Home/>
      </PusherContext.Provider>
    )
    
    expect(await screen.findByText('200 Dr.')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
