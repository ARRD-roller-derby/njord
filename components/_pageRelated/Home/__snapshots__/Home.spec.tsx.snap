// Vitest Snapshot v1

exports[`<Home /> > Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_lmcue_1"
    data-color-mode="dark"
  >
    <header
      class="_header_lmcue_18"
    >
      <header
        class="_header_1yhnh_1"
      >
        <div
          class="_appName_1yhnh_7"
        >
          <a
            class="_title_1045q_1"
            title="retour à l'accueil"
          >
            Njörd
          </a>
          <div
            class="_link_1ld19_1"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="flêche"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div>
              Njörd
            </div>
          </div>
        </div>
        <div
          class="_appName_1yhnh_7"
        >
          <div
            class="_count_snjyq_1"
          >
            200 Dr.
          </div>
          <a
            class="_container_1ssp7_1"
            data-active="false"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="clôche"
                class="_bell_1ssp7_11"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div
              class="_count_13krl_1"
            >
               10
            </div>
          </a>
          <div
            class="_container_1cugo_1"
          >
            <div
              class="_avatar_1cugo_12"
              data-ignore="true"
              data-testid="avatar"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2730%27%20height=%2730%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="/public/static/images/profile.webp"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
          </div>
        </div>
      </header>
      <div
        class="Toastify"
      />
    </header>
    <div
      class="_menu_lmcue_22"
    />
    <main
      class="_main_lmcue_12"
    >
      <div
        class="_box_lmcue_31"
      >
        <div
          class="_container_k6biv_1"
        >
          <div
            class="_container_1geaq_1"
          >
            <div
              class="_filter_1geaq_15"
            >
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-115734p-Control"
                >
                  <div
                    class=" css-kqoj1a-ValueContainer"
                  >
                    <div
                      class=" css-wcgefx-singleValue"
                    >
                      tous
                    </div>
                    <div
                      class=" css-nkhigo-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-2-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1yctulr-IndicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-1gaipyb-DropdownIndicator"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="_noResult_1geaq_8"
            >
              <div
                class="_wheel_1ukrt_5"
              >
                <div
                  class="_border_wheel_1ukrt_1"
                >
                  <div
                    class="_inside_wheel_1ukrt_2"
                  >
                    <div
                      class="_whole_wheel_1ukrt_7"
                    />
                    <div
                      class="_screw_1ukrt_4"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
